### ISSUE TITLE
Sound effects and music don't work in Safari with HTML5 target

### ISSUE THREAD

**Issue by brettchalupa** [HaxeFlixel/flixel#2687]
- **Haxe version:** 4.2.5
- **Flixel version:** 4.11.0
- **OpenFL version:** 9.2.0
- **Lime version:** 8.0.0
- **Affected targets:**
  - html5, Safari only

________________________________

**Code snippet reproducing the issue**:

```haxe
package;

import flixel.FlxG;
import flixel.FlxState;

class PlayState extends FlxState
{
	override public function create()
	{
		super.create();
	}

	override public function update(elapsed:Float)
	{
		super.update(elapsed);

		if (FlxG.mouse.justPressed)
		{
			FlxG.sound.play(AssetPaths.click__wav);
		}
	}
}
```

Here's that click wav file, although I know it happens with other files: https://github.com/brettchalupa/LonelyPaddle/blob/main/assets/sounds/click.wav

Here's a zip file of a project that contains the file and code:
[HaxeFlixel-HTML5-Safari-Sound-Demo.zip](https://github.com/HaxeFlixel/flixel/files/10123723/HaxeFlixel-HTML5-Safari-Sound-Demo.zip)

________________________________

**Observed behavior:** sound fails to play and an error is logged in the console.

```
TypeError: null is not an object (evaluating 'this.__buffer.get_src().duration')
```

Here's a stack trace (I couldn't highlight and copy the text strangely enough):

<img width="697" alt="image" src="https://user-images.githubusercontent.com/928367/204812571-3cb6a7e0-49c2-49d7-a13e-31a2aa187d04.png">

The `Sound.hx` file does appear to be coming from OpenFL, so let me know if I need to submit an issue for that project!

This only happens in Safari. I am running MacOS 12.6 with Safari 16.0. I know that sounds and music used to work in Safari at least a year ago. Everything works totally fine in Firefox and Chrome.

This happens regardless of the sound effect being a .wav or a .ogg.

**Expected behavior:** sound effect plays on click

**Comment by brettchalupa** [comment]
@Geokureli I'm having a tough time writing the needed OpenFL code to repro, but here's what I've got if you (or anyone else) can lend some guidance in how to test this properly. This doesn't work or do anything on click. 

``` haxe
package;

import openfl.events.MouseEvent;
import openfl.media.Sound;
import openfl.display.Sprite;

class Main extends Sprite
{
	public function new()
	{
		super();

		this.addEventListener(MouseEvent.MOUSE_DOWN, function(e:MouseEvent) {
			trace("hi");
			Sound.loadFromFile("assets/click.wav").onComplete(function(s) {
				s.play();
			});
		});
	}
}
```

[OpenFLSoundTest-Failed.zip](https://github.com/HaxeFlixel/flixel/files/10126582/OpenFLSoundTest-Failed.zip)

**Comment by Geokureli** [comment]
try `stage.addEventListener` instead of `this`

**Comment by brettchalupa** [comment]
@Geokureli thanks! that was an easy change. 😄 It works in Firefox and Chrome but it fails in Safari with this console error:

``` console
[lime.utils.AssetLibrary] WARNING: Could not load "assets/click.wav"
10SoundTestFL.js:4206
```

Shall I open up an issue with OpenFL then? Seems like it's stemming from there, although the error is different than what I posted originally.

**Comment by Geokureli** [comment]
Yeah, there's not much we can do, here. you might want to try an mp3 instead of wav, for the time being. No idea when openfl/lime will get to this.

**Comment by brettchalupa** [comment]
Thanks, sounds good.

I ended up logging the issue in Lime because that's where the error was coming from. Hoping we can get to the bottom of it: https://github.com/openfl/lime/issues/1607

I'll do some testing with mp3 files and report back on how that goes. 👍

**Comment by brettchalupa** [comment]
I did some testing with the converting of my WAV audio files to MP3, and it automatically loads them in Safari. So while WAV support in OpenFL would be nice, I'm gonna just close this. At least it's documented if people run into this issue. The solution is to have MP3 and WAV versions of your sound effects and OGG and MP3 versions of your music.

I used ffmpeg to convert to MP3, and it's pretty painless:

``` console
ffmpeg -vn -i assets/sounds/star.wav -acodec libmp3lame assets/sounds/star.mp3
```
